helm/kube-prometheus/templates/configmap.yaml

```diff
@@ -14,9 +14,9 @@ data:
 {{- $root := . }}
 {{- range $key, $val := .Values.prometheusRules }}
   {{ $key }}: |-
-{{ $val | indent 4}}
+{{ tpl $val $root | indent 4}}
```

Comment:
Does this change break existing rules?

Reply:
It hasn't broken anything in our own testing here, however, it would be great if there was a way to actively certify this.

Reply:
Could you run the helm install command with ```--dry-run --debug``` and compare the output  of both?

Reply:
I believe the output looks the same, other than debug output and some ordering.

The old files were produced by:

helm install ~/scratch/current/prometheus-operator/helm/kube-prometheus/ --name kube-prometheus-test --set rbacEnable=true --namespace monitoring --dry-run --debug  >/tmp/current-output.txt


The new files were produced by:

helm install ~/scratch/fork/prometheus-operator/helm/kube-prometheus/ --name kube-prometheus-test --set rbacEnable=true --namespace monitoring --dry-run --debug  >/tmp/fork-output.txt


The diff looks good:

14:28:36|dhart|~/scratch|prod|cluster1.pstg-dev.net||40 mins$ diff /tmp/current-output.txt /tmp/fork-output.txt
1c1
< [debug] Created tunnel using local port: '53097'
---
> [debug] Created tunnel using local port: '53121'
3c3
< [debug] SERVER: "localhost:53097"
---
> [debug] SERVER: "localhost:53121"
6c6
< [debug] CHART PATH: /Users/dhart/scratch/current/prometheus-operator/helm/kube-prometheus
---
> [debug] CHART PATH: /Users/dhart/scratch/fork/prometheus-operator/helm/kube-prometheus
10c10
< RELEASED: Fri Mar 23 14:27:31 2018
---
> RELEASED: Fri Mar 23 14:28:13 2018



And finally, validating the diff of these folders matches the PR:

14:31:22|dhart|~/scratch|prod|cluster1.pstg-dev.net||38 mins$ diff -r /Users/dhart/scratch/current/prometheus-operator/helm/kube-prometheus /Users/dhart/scratch/fork/prometheus-operator/helm/kube-prometheus
diff -r /Users/dhart/scratch/current/prometheus-operator/helm/kube-prometheus/requirements.lock /Users/dhart/scratch/fork/prometheus-operator/helm/kube-prometheus/requirements.lock
36c36
< generated: 2018-03-23T14:27:02.523494782-07:00
---
> generated: 2018-03-23T14:28:05.479708584-07:00
diff -r /Users/dhart/scratch/current/prometheus-operator/helm/kube-prometheus/templates/configmap.yaml /Users/dhart/scratch/fork/prometheus-operator/helm/kube-prometheus/templates/configmap.yaml
17c17
< {{ $val | indent 4}}
---
> {{ tpl $val $root | indent 4}}
22c22
< {{ end }}
\ No newline at end of file
---
> {{ end }}